packages/atomic/src/components/search/atomic-segmented-facet-scrollable/atomic-segmented-facet-scrollable.tsx

```diff
@@ -0,0 +1,189 @@
+  InitializeBindings,
+} from '../../../utils/initialization-utils';
+import {Bindings} from '../atomic-search-interface/atomic-search-interface';
+import {
+  buildSearchStatus,
+  SearchStatus,
+  SearchStatusState,
+} from '@coveo/headless';
+import {Hidden} from '../../common/hidden';
+
+type ArrowDirection = 'right' | 'left';
+
+/**
+ * @internal
+ * The 'atomic-segmented-facet-scrollable' component wraps around one or several 'atomic-segmented-facet' to provide horizontal scrolling capabilities.
+ * @part scrollable-container - The wrapper for the entire component including the horizontal-scroll container and the arrow buttons.
+ * @part horizontal-scroll - The scrollable container for the segmented facets.
+ * @part left-arrow-box - The left arrow box containing both the left arrow button and the fade.
+ * @part right-arrow-box - The right arrow box containing both the right arrow button and the fade.
+ * @part arrow-button - The arrow button used to scroll left or right.
+ * @part fade - The white to transparent gradient.
+ */
+
+@Component({
+  tag: 'atomic-segmented-facet-scrollable',
+  styleUrl: 'atomic-segmented-facet-scrollable.pcss',
+  shadow: true,
+})
+export class AtomicSegmentedFacetScrollable implements InitializableComponent {
+  @InitializeBindings()
+  public bindings!: Bindings;
+  public searchStatus!: SearchStatus;
+  @BindStateToController('searchStatus')
+  @State()
+  public searchStatusState!: SearchStatusState;
+  @State()
+  public error!: Error;
+
+  private horizontalScroll?: HTMLDivElement;
+  @State() private hideLeftArrow = true;
+  @State() private hideRightArrow = false;
+  private ro!: ResizeObserver;
+
+  public initialize() {
+    this.searchStatus = buildSearchStatus(this.bindings.engine);
+    console.log('init');
+  }
+
+  componentDidLoad() {
+    this.ro = new ResizeObserver(() => {
+      this.handleScroll();
+    });
+    this.ro.observe(this.horizontalScroll as HTMLDivElement);
+  }
+
+  disconnectedCallback() {
+    this.ro.disconnect();
+  }
+
+  @Listen('mousewheel')
+  handleScroll() {
+    if (!this.horizontalScroll) {
+      return;
+    }
+
+    const isOverflowing =
+      this.horizontalScroll.clientWidth < this.horizontalScroll.scrollWidth;
+    const isLeftEdge = this.horizontalScroll?.scrollLeft <= 0;
+    const isRightEdge =
+      this.horizontalScroll.scrollLeft >=
+      this.horizontalScroll.scrollWidth - this.horizontalScroll.clientWidth;
+
+    if (!isOverflowing) {
+      this.hideLeftArrow = true;
+      this.hideRightArrow = true;
+    }
+    if (isLeftEdge) {
+      this.hideLeftArrow = true;
+    } else {
+      this.hideLeftArrow = false;
+    }
+    if (isRightEdge) {
+      this.hideRightArrow = true;
+    } else {
+      this.hideRightArrow = false;
+    }
+  }
+
+  private slideHorizontally(direction: ArrowDirection) {
+    const container = this.horizontalScroll;
+    if (!container) {
+      return;
+    }
+
+    const containerWidth = container.clientWidth;
```

Comment:
if there is ever a case where container in undefined, like you expect on line 104, this will break but in general, I think you should choose one approach - to save stuff as variables or not, for instance on line 104 you use `container.clientWidth` and the on line 117 you use `containerWidth`. I'd suggest sticking to §`container.clientWidth`, unless you then rename things to `width`, `scrollWidth`, `scrollLeft` (they are a bit shorter and therefore easier to read).

Of course my comment above doesn't apply if you were trying to save the original value before changing it (which would explain the mix of `containerWidth` and `container.clientWidth`) but then I'd rename the variables to 'original.....`

Reply:
Sure !